contracts/src/ReferralCampaign.sol

```diff
@@ -131,10 +131,11 @@ contract ReferralCampaign is Ownable,Initializable {
 
     //@dev Function to add a Referrer
     function addReferrer(address signal, uint256 root, uint256 nullifierHash, uint256[8] calldata proof) public {
-        verifyAndExecute(signal, root, nullifierHash, proof);
         //@dev Instead of require, verification to be done by worldcoin
         require(numReferralsByReferrer[msg.sender] == 0, "Referrer already registered.");
 
+        // verifyAndExecute(signal, root, nullifierHash, proof);
```

Comment:
@mani99brar @thisisashugupta notice that I'm moving this down. We should only check humanness as the last step.